doc: add CONTRIBUTING.md & update README.md (#39)
vlang/go2v: CONTRIBUTING.md

# Special thanks for your interest in contributing, you are more than welcome!

> ## The translation from Go to V follow those steps:

Reviewer: ```suggestion
## The translation from Go to V follows those steps:
```

> ## The translation from Go to V follow those steps:

Reviewer: ```suggestion
## The translation from Go to V follows those steps:
```

> 1. Go2V CLI app

Reviewer: ```suggestion
1. Start the process
```

   - `go2v.v`

> 2. Get the Go code (path handling etc.)

Reviewer: ```suggestion
2. The transpiler core starts by getting the Go code (path handling etc.)
```

	- `transpiler/go2v.v`
3. Generate Go AST
	- `transpiler/get_ast.go`

> 	- `transpiler/go2v.v`

Reviewer: Not necessary to show this again.

Author: Okay maybe `transpiler/go2v.v` isn't useful here but IMHO `transpiler/get_ast.go` is useful

Reviewer: I only meant the 2nd line.  :-)

Author: You want to remove "3. Generate Go AST"?

Reviewer: I want to remove line 11:  "- `transpiler/go2v.v`" - that's the part that doesn't need to be repeated.  It's the second line in that section.

Author: I don't really see why that's an issue, `transpiler/go2v.v` is effectively used in both steps

Reviewer: Only to call the next routine.  It doesn't do the AST generation.  However, it doesn't really hurt anything to leave it, and if that's what you want to do, I have no objection.

4. Parse Go AST
   - Tokenize Go AST
      - `transpiler/tokenizer.v`
   - Transform the list of tokens into a tree
      - `transpiler/tree_constructor.v`

> 5. Transform the parsed AST into a VAST struct

Reviewer: ```suggestion
5. Transform the parsed Go AST into a V AST struct
```

   - `transpiler/ast_constructor.v`
   - `transpiler/utils.v`
6. Adapt remaining Go-ish things to V-ish things
   - `transpiler/v_style.v`
7. Transform the V AST struct into V Code
   - `transpiler/v_file_constructor.v`

## When adding support for new cases you must add corresponding test(s).

*In this section replace `go2v` with `v run .` or compile Go2V first.*

> Tests are located in `tests` subdirectories, each test is composed of a `.go` file, the input, and a `.vv` file the supposed output.

Reviewer: ```suggestion
Tests are located in `tests` subdirectories, each test is composed of a `.go` file, the input, and a `.vv` file containing the expected output.
```

> You can run the test suit using `go2v test` or specific one using `go2v test test_name`.

Reviewer: ```suggestion
You can run the test suite using `go2v test` or specific one using `go2v test test_name`.
```

> You can run the test suite using `go2v test` or specific one using `go2v test test_name`.

Reviewer: ```suggestion
You can run the test suite using `go2v test` or a specific test using `go2v test test_name`.
```

> - To create a test simply run the `go2v test -create test_name` command, it'll create a a dummy Go file at `tests/test_name/test_name.go`

Reviewer: ```suggestion
- To create a test simply run `go2v test -create test_name` command, it'll create a dummy Go file at `tests/test_name/test_name.go`
```
